### PR TITLE
8 0 16 encryption reconsiliation encryption part 2

### PR DESCRIPTION
**Pull request by robgolebiowski**

### CODE REVIEW

File: mysql-test/collections/disabled.def [hunk 1]
```diff
@@ -173,6 +173,8 @@ tokudb.rpl.rpl_partition_tokudb : BUG#0 exchange partition is not supported
 
 # encryption suite tests
 encryption.innodb-bad-key-change2 : BUG#5322 percona https://jira.percona.com/browse/PS-5322
+# PS-5772 8.0.16 - encryption reconciliation - fix after merge
```
Comment (laurynas-biveinis) [inline mysql-test/collections/disabled.def]:
```suggestion
```

File: mysql-test/collections/disabled.def [hunk 2]
```diff
@@ -173,6 +173,8 @@ tokudb.rpl.rpl_partition_tokudb : BUG#0 exchange partition is not supported
...
+encryption.innodb-bad-key-change2 : BUG#5322 percona https://jira.percona.com/browse/PS-5322
```
Comment (laurynas-biveinis) [inline mysql-test/collections/disabled.def]:
```suggestion
encryption.innodb-bad-key-change2 : BUG#5772 percona encryption reconciliation - fix after merge https://jira.percona.com/browse/PS-5772
```

File: mysql-test/suite/encryption/r/explicit_encryption.result [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#
+# PS-5772 8.0.16 encryption reconciliation
+# Check if explicitt ENCRYPTION='N' is always shown in SHOW CREATE TABLE
```
Comment (laurynas-biveinis) [inline mysql-test/suite/encryption/r/explicit_encryption.result]:
```suggestion
# Check if explicit ENCRYPTION='N' is always shown in SHOW CREATE TABLE
```

File: mysql-test/suite/encryption/t/explicit_encryption.test [hunk 4]
```diff
@@ -0,0 +1,37 @@
+--echo #
+--echo # PS-5772 8.0.16 encryption reconciliation
+--echo # Check if explicitt ENCRYPTION='N' is always shown in SHOW CREATE TABLE
```
Comment (laurynas-biveinis) [inline mysql-test/suite/encryption/t/explicit_encryption.test]:
```suggestion
--echo # Check if explicit ENCRYPTION='N' is always shown in SHOW CREATE TABLE
```

File: mysql-test/suite/encryption/t/explicit_encryption.test [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+--echo # for innodb_file_per_table tables. It should not be printed for any
+--echo # different table (in general or system tablespace). Please note we test
+--echo # explicit ENCRYPTION='N' and not the rules of printing this clause in
+--echo # SHOW CREATE TABLE according to WL#12261.
+--echo #
+--source include/have_debug.inc
```
Comment (laurynas-biveinis) [inline mysql-test/suite/encryption/t/explicit_encryption.test]: Needed?

File: sql/sys_vars.cc [hunk 6]
```diff
@@ -6961,17 +6961,15 @@ static bool check_set_default_table_encryption_access(
...
-      var->save_result.ulonglong_value)
-    return 0;
-
+static const char *default_table_encryption_type_names[] = {
+    "OFF",
+    "ON",
+    "KEYRING_ON",
+    "ONLINE_TO_KEYRING",
+    "ONLINE_FROM_KEYRING_TO_UNENCRYPTED",
+    0};
```
Comment (laurynas-biveinis) [inline sql/sys_vars.cc]:
```suggestion
    nullptr};
```

That's the ASan fix right?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: sql/sys_vars.cc (hunk 6 above)
Comment (robgolebiowski) [reply sql/sys_vars.cc]: I hope so - noticed it before you asked me about the ASAN issue, but it was most definitely causing out of scope memory accesses.

**Comment by robgolebiowski** [comment]
https://ps80.cd.percona.com/job/percona-server-8.0-param/261/ - keyring_vault.table_encrypt_3 and keyring_vault.table_encrypt_3_directory should have been fixed by now, also recorded result file for explicit_encryption test.

**Pull request merged**